### PR TITLE
GRW-810/ Offer Page - Update layout to support multiple start date fields

### PR DESCRIPTION
**Pull request by marian-buc**
## What?
Modified StartDate component to be able to create DateForms containing up to 4 input fields, grouped 2 by 2.

## Why?
To be able to select Start Dates for Travel and House insurance coverages

**Ticket(s): [GRW-810]**

image
image

[GRW-810]: https://hedvig.atlassian.net/browse/GRW-810?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/client/pages/Offer/Introduction/Sidebar/StartDate.tsx [hunk 1]
```diff
@@ -395,7 +406,7 @@ export const StartDate = ({
           />
         ) : (
           <>
-            {quotes.map((quote) => (
+            {quotes.map((quote, index, arr) => (
```
Comment (marian-buc) [inline src/client/pages/Offer/Introduction/Sidebar/StartDate.tsx:409]: Test with ex.: `quotes.concat([quotes[0]]).map`

**Review by simonauner — DISMISSED** [review]
I would probably have implemented a ButtonGroup component to handle this, but this seems simpler and more straight-forward for the specific situation 👍

**Comment by marian-buc** [comment]
Yep, I agree it could have been done in a neater way, without manual control over the isSplit prop, but I would have to refactor pretty much the whole StartDate component, I preferred to just keeps what's good and change the rest

File: src/client/pages/Offer/Introduction/Sidebar/StartDate.tsx [hunk 2]
```diff
@@ -31,18 +31,23 @@ import { CancellationOptions } from './CancellationOptions'
 const DateFormsWrapper = styled.div`
   display: flex;
   flex-direction: row;
+  flex-wrap: wrap;
 `
+type SplitData = 'left' | 'right' | false
```
Comment (robinandeer) [inline src/client/pages/Offer/Introduction/Sidebar/StartDate.tsx:36]: Isn't it clearer? Or something similar

```suggestion
type SplitData = 'left' | 'right' | 'full'
```

File: src/client/pages/Offer/Introduction/Sidebar/StartDate.tsx [hunk 3]
```diff
@@ -395,26 +406,46 @@ export const StartDate = ({
...
-                }
-                loading={loadingQuoteIds.includes(quote.id)}
-              />
-            ))}
+            {quotes.map((quote, index, arr) => {
+              const isArrayLengthEven = arr.length % 2 === 0
+              const isItemIndexEven = index % 2 === 0
+              const isMultiQuote = quoteBundleSelector.isMultiQuote(
+                selectedBundle,
+              )
```
Comment (robinandeer) [inline src/client/pages/Offer/Introduction/Sidebar/StartDate.tsx:414]: We don't need to check this anymore, right?

`isExpandedFirstItem === true` works for 1, 3, 5 items :)

**Review by robinandeer — APPROVED** [review]

File: src/client/pages/Offer/Introduction/Sidebar/StartDate.tsx [hunk 4]
```diff
@@ -33,12 +33,12 @@ const DateFormsWrapper = styled.div`
   flex-direction: row;
   flex-wrap: wrap;
 `
-type SplitData = 'left' | 'right' | false
+type TFieldLayout = 'left' | 'right' | 'full'
```
Comment (robinandeer) [inline src/client/pages/Offer/Introduction/Sidebar/StartDate.tsx:36]: Nice!

**Review by robinandeer — APPROVED** [review]
Great improvement!

**Pull request merged**